connector/servicegraphconnector/connector.go

```diff
@@ -155,33 +154,12 @@ func newConnector(set component.TelemetrySettings, config component.Config) *ser
 	}
 }
 
-type getExporters interface {
-	GetExporters() map[component.DataType]map[component.ID]component.Component
-}
-
-func (p *serviceGraphConnector) Start(_ context.Context, host component.Host) error {
+func (p *serviceGraphConnector) Start(_ context.Context, _ component.Host) error {
 	p.store = store.NewStore(p.config.Store.TTL, p.config.Store.MaxItems, p.onComplete, p.onExpire)
 
 	if p.metricsConsumer == nil {
```

Comment:
IIUC after this change `p.config.MetricsExporter` is ignored? Should we deprecate this option as part of this PR?

Reply:
That's an interesting observation: following the code, I thought this code was dead, as there would always be a metricsConsumer being set, here:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/73ccd5bb344c0db7b4d9131eb90ae26ef055a522/connector/servicegraphconnector/factory.go#L72-L76

I would assume that the factory would always pass a non-nil metrics consumer, but perhaps I'm wrong?

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/73ccd5bb344c0db7b4d9131eb90ae26ef055a522/connector/servicegraphconnector/factory.go#L52-L59

If I'm right, I'll log a warning in case that option is used, as it would be considered a configuration error, but I don't want to break current users, even though their value for the metrics exporter is nop at the moment.

Reply:
> I would assume that the factory would always pass a non-nil metrics consumer, but perhaps I'm wrong?

Yeah I think that's the case, so this code is indeed dead.

>  I'll log a warning in case that option is used, as it would be considered a configuration error, but I don't want to break current users, even though their value for the metrics exporter is nop at the moment.

Makes sense, no need to break users, but I would want to deprecate the option so that we can eventually remove it

Reply:
Alright, I added a new commit to this PR with this change.